### PR TITLE
Enable AZSLc and Shader Compilation on Linux

### PR DESCRIPTION
**Pull request by spham-amzn**
* Add azslc package 3P definition for download
* Add AzslcGeader.azsli nad PlatformHeader.hlsli for Linux/Null and Linux/Vulkan (Copy of Android/Vulkan)
* Enable trait PAL_TRAIT_BUILD_ATOM_ASSET_SHADER_SUPPORTED for Linux
* Add platform builder file dependencies
* Updated AZ_TRAIT_ATOM_SHADERBUILDER_AZSLC to use new 3P linux version of azslc
* Add AtomShaderCapabilities.json, PlatformLimits.azasset for Linux (Copy from Android)
* Fix paths/values for Vulkan_Traits_Linux.h
* Enable trait PAL_TRAIT_BUILD_ATOM_ASSET_SHADER_SUPPORTED for Linux

### CODE REVIEW

**Review by amzn-pratikpa — APPROVED** [review]

File: Gems/Atom/Feature/Common/Assets/Config/Platform/Linux/AtomShaderCapabilities.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        "DescriptorCounts":
+        {
+            "Sets": 80,
+            "Spaces": 8,
+            "Samplers": 8,
+            "Textures": -1,
+            "Buffers": -1
+        },
+        "MaxSpaces": 4
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/Config/Platform/Linux/AtomShaderCapabilities.json:15]: Dont add this file. This number enforcers SRG merging due to restrictions on the number oof SRGs allowed on Android.

File: Gems/Atom/Asset/Shader/Code/AZSL/Platform/Linux/Vulkan/AzslcHeader.azsli [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ */
+
+/*
+* The shader build pipeline has 2 preprocess stages. The first one happens 
+* after appending this header and pre-processing with mcpp ahead of azslc.
+*/
+
+//Used for normal map sampling
+#define AZ_TRAIT_ASTC_COMPRESSION 1
```
Comment (fabioanderegg) [inline Gems/Atom/Asset/Shader/Code/AZSL/Platform/Linux/Vulkan/AzslcHeader.azsli:15]: I don't have that in my branch, I copied the file from Windows, this looks like it comes from Android which needs that define?

File: Gems/Atom/RHI/Vulkan/Code/Source/Platform/Linux/Vulkan_Traits_Linux.h [hunk 3]
```diff
@@ -7,9 +7,9 @@
...
 #pragma once
 
-#define AZ_TRAIT_ATOM_SHADERBUILDER_DXC "Builders/DirectXShaderCompilerAz/bin/dxc"
+#define AZ_TRAIT_ATOM_SHADERBUILDER_DXC "Builders/DirectXShaderCompiler/bin/dxc"
 #define AZ_TRAIT_ATOM_VULKAN_DISABLE_DUAL_SOURCE_BLENDING 0
-#define AZ_TRAIT_ATOM_VULKAN_DLL ""
-#define AZ_TRAIT_ATOM_VULKAN_DLL_1 ""
+#define AZ_TRAIT_ATOM_VULKAN_DLL "libvulkan.so"
+#define AZ_TRAIT_ATOM_VULKAN_DLL_1 "libvulkan.so.1"
 #define AZ_TRAIT_ATOM_VULKAN_LAYER_LUNARG_STD_VALIDATION_SUPPORT 0
```
Comment (fabioanderegg) [inline Gems/Atom/RHI/Vulkan/Code/Source/Platform/Linux/Vulkan_Traits_Linux.h:14]: Validation layers are working for me, so we could activate them here.

**Review by fabioanderegg — COMMENTED** [review]
Comparing this to the equivalent commits on my branch, it's mostly the same: https://github.com/fabioanderegg/o3de/tree/linux-vulkan2

Some notes:
* Should we also add the new targets for Linux Null and Vulkan in Gems/Atom/Asset/Shader/Code/Source/Platform/Linux/platform_builders_linux.cmake to
Gems/Atom/Asset/Shader/Code/Source/Platform/Mac/platform_builders_mac.cmake and
Gems/Atom/Asset/Shader/Code/Source/Platform/Windows/platform_builders_windows.cmake?
* I've also activated `PAL_TRAIT_BUILD_HOST_GUI_TOOLS` in `cmake/Platform/Linux/PAL_linux.cmake`, but that can probably wait for another pull request.

**Review by galibzon — APPROVED** [review]

File: Gems/Atom/Asset/Shader/Code/AZSL/Platform/Linux/Vulkan/AzslcHeader.azsli (hunk 2 above)
Comment (moudgils) [reply Gems/Atom/Asset/Shader/Code/AZSL/Platform/Linux/Vulkan/AzslcHeader.azsli:15]: yes. We dont need to enable AZ_TRAIT_ASTC_COMPRESSION for Linux as you should be able to use BCn compressions similar to PC

File: Gems/Atom/RHI/Vulkan/Code/Source/Platform/Linux/Vulkan_Traits_Linux.h (hunk 3 above)
Comment (moudgils) [reply Gems/Atom/RHI/Vulkan/Code/Source/Platform/Linux/Vulkan_Traits_Linux.h:14]: +1

File: Gems/Atom/Feature/Common/Assets/Config/Platform/Linux/AtomShaderCapabilities.json (hunk 1 above)
Comment (spham-amzn) [reply Gems/Atom/Feature/Common/Assets/Config/Platform/Linux/AtomShaderCapabilities.json:15]: Will Remove

File: Gems/Atom/Asset/Shader/Code/AZSL/Platform/Linux/Vulkan/AzslcHeader.azsli (hunk 2 above)
Comment (spham-amzn) [reply Gems/Atom/Asset/Shader/Code/AZSL/Platform/Linux/Vulkan/AzslcHeader.azsli:15]: Will remove this define

File: Gems/Atom/RHI/Vulkan/Code/Source/Platform/Linux/Vulkan_Traits_Linux.h (hunk 3 above)
Comment (spham-amzn) [reply Gems/Atom/RHI/Vulkan/Code/Source/Platform/Linux/Vulkan_Traits_Linux.h:14]: Will update to 1

**Comment by spham-amzn** [comment]
> Comparing this to the equivalent commits on my branch, it's mostly the same: https://github.com/fabioanderegg/o3de/tree/linux-vulkan2
> 
> Some notes:
> 
>     * Should we also add the new targets for Linux Null and Vulkan in Gems/Atom/Asset/Shader/Code/Source/Platform/Linux/platform_builders_linux.cmake to
>       Gems/Atom/Asset/Shader/Code/Source/Platform/Mac/platform_builders_mac.cmake and
>       Gems/Atom/Asset/Shader/Code/Source/Platform/Windows/platform_builders_windows.cmake?
> 
>     * I've also activated `PAL_TRAIT_BUILD_HOST_GUI_TOOLS` in `cmake/Platform/Linux/PAL_linux.cmake`, but that can probably wait for another pull request.

1. Yeah, i'll add them there for consistency
2. This might be a dead trait as I didnt find what it was exactly gating.

**Review by moudgils — APPROVED** [review]

**Pull request merged**